pkg/instance/updates.go

```diff
@@ -0,0 +1,248 @@
+package instance
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/cozy/cozy-stack/pkg/apps"
+	"github.com/cozy/cozy-stack/pkg/config"
+	"github.com/cozy/cozy-stack/pkg/logger"
+	"github.com/cozy/cozy-stack/pkg/utils"
+	multierror "github.com/hashicorp/go-multierror"
+	"github.com/robfig/cron"
+)
+
+const numUpdaters = 50
+
+var log = logger.WithNamespace("updates")
+var globalUpdating = make(chan struct{}, 1)
+
+func init() {
+	globalUpdating <- struct{}{}
+}
+
+type updateCron struct {
+	stopped  chan struct{}
+	finished chan struct{}
+}
+
+// StartUpdateCron starts the auto update process which launche a full auto
+// updates of all the instances existing.
+func StartUpdateCron() (utils.Shutdowner, error) {
+	autoUpdates := config.GetConfig().AutoUpdates
+	if !autoUpdates.Activated {
+		return utils.NopShutdown, nil
+	}
+
+	u := &updateCron{
+		stopped:  make(chan struct{}),
+		finished: make(chan struct{}),
+	}
+
+	scheduleValue := autoUpdates.Schedule
+	if scheduleValue == "" {
+		scheduleValue = "@midnight"
```

Comment:
I thought `""` means that the auto update is disabled, https://github.com/cozy/cozy-stack/pull/948/files#diff-091ace02b49a46f55f2586cfec89a160R306